### PR TITLE
server: update dispatch behavior on mysql command SLEEP

### LINKED ISSUE #38612: Update the handling of COM_SLEEP
## Enhancement

The current tidb (commit `41518c9e02`) handles `COM_SLEEP` specially and doesn't take it as an unknown command.

```go
// server/conn.go:(*clientConn).dispatch()
	case mysql.ComSleep:
		// TODO: According to mysql document, this command is supposed to be used only internally.
		// So it's just a temp fix, not sure if it's done right.
		// Investigate this command and write test case later.
		return nil
```

However, the current mysql (commit `a246bad7`) takes it as an unknown command and raise an error. 

```c++
// sql/sql_parse.cc:dispatch_command()
    case COM_SLEEP:
    case COM_CONNECT:         // Impossible here
    case COM_TIME:            // Impossible from client
    case COM_DELAYED_INSERT:  // INSERT DELAYED has been removed.
    case COM_END:
    default:
      my_error(ER_UNKNOWN_COM_ERROR, MYF(0));
      break;
```

Should we follow the mysql implementation to return unknown error when handling this command?

### PR DESCRIPTION
**Pull request by igxlin**
The handling is now matching the MySQL. It will raise error UNKNOWN COMMAND when receiving command COM_SLEEP from clients.

### What problem does this PR solve?

Issue Number: close #38612

Problem Summary:

### What is changed and how it works?

The COM_SLEEP is not taken as an unknown command when TIDB receives it. It doesn't match the MySQL 5.7 implementation. This PR is to fix it and make the behavior matches. 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by igxlin** [comment]
/cc @xhebox 

PR is created to close #38612 . Please help review at your convenience. Thanks.

File: server/conn_test.go [hunk 1]
```diff
@@ -349,12 +349,6 @@ func TestDispatch(t *testing.T) {
 	userData = append(userData, 0x0)
 
 	inputs := []dispatchInput{
-		{
-			com: mysql.ComSleep,
-			in:  nil,
-			err: nil,
```
Comment (xhebox) [inline server/conn_test.go:355]: Should be able to assert some errors

Comment (igxlin) [reply server/conn_test.go:355]: @xhebox Updated. Please help review. 

Commit: https://github.com/pingcap/tidb/pull/38645/commits/16a5c7d895ca92a554b1563ded33d7e8b1f36345

```go
inputs := []dispatchInput{
        {
            com: mysql.ComSleep,
            in:  nil,
            err: mysql.NewErrf(mysql.ErrUnknown, "command %d not supported now", nil, mysql.ComSleep),
            out: nil,
        },
```

**Review by xhebox — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**